release/Magarena/scripts/Maulfist_Revolutionary.groovy

```diff
@@ -1,10 +1,24 @@
+def TARGET_PERMANENT_OR_PLAYER = new MagicTargetChoice() {
+    new MagicTargetFilterImpl {
+        @Override
+        public boolean acceptType(final MagicTargetType targetType) {
+            return targetType == MagicTargetType.Player || targetType == MagicTargetType.Permanent;
+        }
+        @Override
+        public boolean accept(final MagicSource source,final MagicPlayer player,final T target) {
+            return true;
+        }
+    },
+    "target permanent or player"
+}
+
 [
     new EntersBattlefieldTrigger() {
         @Override
         public MagicEvent executeTrigger(final MagicGame game,final MagicPermanent permanent, final MagicPayedCost payedCost) {
             return new MagicEvent(
                 permanent,
-                new MagicTargetChoice("target permanent or player"),
+                TARGET_PERMANENT_OR_PLAYER,
                 this,
                 "For each kind of counter on target permanent or player,\$ " +
```

Comment:
Please move the target indicator `\$` before the comma, otherwise it looks quite strange in the UI, eg "For each kind of counter on target permanent or player, (Your Name) give that permanent ..."

It should be "For each kind of counter on target permanent or player (Your Name), give that permanent ..." when the indicator is put before the comma.